python-package/lightgbm/basic.py

```diff
@@ -3706,6 +3706,11 @@ def __getstate__(self) -> Dict[str, Any]:
         return this
 
     def __setstate__(self, state: Dict[str, Any]) -> None:
+        _log_warning(
+            "Starting from version 4.0.0, LightGBM models saved as pickle files might contain "
+            "malicious code and lead to security risks. "
+            "Consider loading models from text files to avoid this issue."
+        )
```

Comment:
I strongly oppose adding this.

It implies that the ability to pickle models was only added in v4.0.0, which is false.